src/requirements/reqs-functions.ts

```diff
@@ -303,9 +306,16 @@ export function computeRequirements(
         .flat();
     },
     getCorrespondingRequirementAndAllRelevantCoursesUnderFulfillmentStrategy: it => it,
-    // TODO: Replace this dummy implementation once we decided how to determine if a requirement is
-    // double-countable. Meanwhile, make it always return true to match the old behavior.
-    allowDoubleCounting: () => true,
+    allowDoubleCounting: requirement => {
+      // All minor requirements are automatically double-countable.
+      if (requirement.sourceType === 'Minor') return true;
+      if (requirement.sourceType === 'Major') {
+        if (majors == null) throw new Error();
```

Comment:
Do we want anything more descriptive thrown even if it should not happen?

Reply:
clarified